### PR TITLE
Add human readable game release file sizes

### PR DESCRIPTION
**Pull request by justinhernandez**
Screen Shot 2020-03-26 at 6 23 20 PM

### CODE REVIEW

**Comment by justinhernandez** [comment]
Need to rework this PR to remove the extra http call and read the value size from mongo

File: app/routes/games/helpers.js [hunk 1]
```diff
@@ -89,6 +148,13 @@ function renderPage(req, res, template, populate=null)
 						: Number(req.query.page)
 					: 1;
 
+			// iterate game releases to add file sizes
+			for (let k = 0; k < game.releases.length; k++) {
+				const commitId = game.releases[k].commitId;
+				const humanFileSize = await releaseSize(game.location, game.slug, commitId)
```
Comment (DylanOrsh) [inline app/routes/games/helpers.js]: So is this iterating and calling for the size on every page load? Has any consideration been given to trying to cache/store the values so we don't have to get them every time? Or is this ultimately fast enough to not be a bother?

File: app/routes/games/helpers.js [hunk 2]
```diff
@@ -33,6 +35,63 @@ function validateRequest(req){
...
+  
+	return ( size / Math.pow(1024, i) ).toFixed(2) * 1 + ' ' + ['B', 'kB', 'MB', 'GB', 'TB'][i];
+}
+
+/**
+ * Read the "content-length" header of the game resource to determine
+ * the byte size
+ * 
+ * @param  {string} host the host where the resource is stored (s3)
+ * @param  {string} commitId commit id of the game release we want to check
```
Comment (902seanryan) [inline app/routes/games/helpers.js]: missing a param comment here for GameSlug

File: app/routes/games/helpers.js (hunk 1 above)
Comment (justinhernandez) [reply app/routes/games/helpers.js]: You are correct to raise these questions. 

Gonna re-work the release model to populate `releaseCompressedSize` (if a value has not been passed) when the object is saved.

**Comment by justinhernandez** [comment]
@DylanOrsh @902seanryan

Since `releaseCompressedSize` is populated by the api, removed the remote HEAD fetch.

File: app/routes/games/helpers.js [hunk 3]
```diff
@@ -89,6 +105,15 @@ function renderPage(req, res, template, populate=null)
 						: Number(req.query.page)
 					: 1;
 
+			// iterate game releases to add file sizes
+			for (let k = 0; k < game.releases.length; k++) {
+        const compressedSize = parseInt(game.releases[k].releaseCompressedSize || 0);
```
Comment (chipbell4) [inline app/routes/games/helpers.js:110]: Looks like there's some tabbing issues here.

File: app/views/games/releases.jade [hunk 4]
```diff
@@ -96,6 +96,10 @@ append gameContent
 											a(href="#{game.repository}/browse?at=refs/heads/#{release.branch.replace('origin/', '')}" data-toggle="tooltip" title="View Branch")=' '+release.branch
 
 								.help-block.updated
+									if release.releaseCompressedSize
```
Comment (chipbell4) [inline app/views/games/releases.jade:99]: Can we also add a block for `releaseUncompressedSize` as well (if it'll fit).

**Pull request merged**